### PR TITLE
added support for PHP 7.2

### PR DESCRIPTION
**Pull request by zemiacsik**
added support for PHP 7.2 and backward compatibility according to [SS 3.7 release notes](https://docs.silverstripe.org/en/3/changelogs/3.7.0/#for-module-authors)

### CODE REVIEW

File: _config.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+// Ensure compatibility with PHP 7.2 ("object" is a reserved word),
+// with SilverStripe 3.6 (using Object) and SilverStripe 3.7 (using SS_Object)
+if (!class_exists('SS_Object')) class_alias('Object', 'SS_Object');
```
Comment (robbieaverill) [inline _config.php]: Can you please format this for PSR-2? It's not something that is enforced in SS3 code but is something we try and follow for any new development =)

```php
if (!class_exists('SS_Object')) {
  class_alias('Object', 'SS_Object');
}
```

Comment (zemiacsik) [reply _config.php]: Of course!

File: _config.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+// Ensure compatibility with PHP 7.2 ("object" is a reserved word),
+// with SilverStripe 3.6 (using Object) and SilverStripe 3.7 (using SS_Object)
+if (!class_exists('SS_Object')) {
+  class_alias('Object', 'SS_Object');
```
Comment (robbieaverill) [inline _config.php]: Sorry, four spaces (nit picking)

Comment (zemiacsik) [reply _config.php]: No problem, I was used to tabs in SS3.

Comment (robbieaverill) [reply _config.php]: We bless them and thank them duly for their long time support and hard work

**Comment by robbieaverill** [comment]
I've noticed that you've created this pull request from one of the "core branches" rather than a branch specifically intended for this pull request. This can cause problems if we ask you to target this PR at a different branch. No action is needed at the moment, this is for your information.

In the future please create a new branch for each individual pull request. See our [guide about contributing PRs](https://docs.silverstripe.org/en/4/contributing/code/#step-by-step-from-forking-to-sending-the-pull-request)

**Comment by zemiacsik** [comment]
Thank you, I will read that. I really appreciate it because I'm not much familiar with contributing on GitHub  yet.
So in this scenario, I forked repo, cloned it, then switched to `branch 2` **and then I should had created a new branch like `2_php7.2support`**?
After that  I would do some changes, commit and make pull request with my new `branch 2_php7.2support` to that `branch 2` or how?

**Comment by robbieaverill** [comment]
No problem, every open source project has it's own requirements too 😄 

Yes, creating a new branch would have been great. It prevents potential problems if I'd wanted to merge this into a branch other than 2, because we would've had to rebase your 2 branch and it's then be out of sync with origin.

Fork, create new branch, commit change in new branch, push, pull request back into original branch (2).

👍

**Pull request merged**